sdk/python/lib/test/runtime/test_stack.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2023, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import traceback
+import pytest
+
+from pulumi.runtime import stack
+from pulumi.runtime.rpc_manager import RPC_MANAGER
+
+
+async def _explode(n = 10):
+    if n == 0:
+        raise Exception("sadness")
+    await _explode(n-1)
+
+
+@pytest.mark.asyncio
+async def test_wait_for_rpcs():
```

Comment:
Maybe add a comment explaining at a high-level that the test is testing that the user's stack trace is propagated.